runtime/lib/ttnn/program.cpp

```diff
@@ -94,21 +98,19 @@ updateLayoutAndDataType(const ::ttnn::Tensor &inputTensor,
   return outputTensor;
 }
 
-// TODO: right now hardcoding tilize/untilize, should determine with tile shape
-// blocked by issue #272
+// TODO(bug #272): right now hardcoding tilize/untilize, should determine with
+// tile shape, blocked by issue
 static void
 run(::tt::target::ttnn::ToMemoryConfigOp const *op, ::ttnn::Device &device,
     std::unordered_map<std::uint32_t, ::ttnn::Tensor *> &liveTensors,
     std::list<::ttnn::Tensor> &tensorPool) {
   const ::ttnn::Tensor &inputTensor = *liveTensors.at(op->in0()->global_id());
-  TT_FATAL(isOnHost(inputTensor) or isOnDevice(inputTensor),
-           "Unsupported storage type {}", inputTensor.storage_type());
+  assert((isOnHost(inputTensor) or isOnDevice(inputTensor)) &&
+         "Unsupported storage type");
```

Comment:
@jnie-TT, unfortunately we cannot use `TT_FATAL`, we just happened to pull in metal's `assert.hpp`, but this is not public API.

Reply:
Sounds good... Thanks for updating this!